Compilation error

While attempting to compile in Fedora 31:

  CC       libdill_la-ipaddr.lo
In file included from ipaddr.c:44:
ipaddr.c: In function ‘dill_ipaddr_remotes’:
dns/dns.h:1009:24: error: lvalue required as unary ‘&’ operand
 1009 | #define dns_opts(...) (&dns_quietinit((struct dns_options)DNS_OPTS_INIT(__VA_ARGS__)))
      |                        ^
ipaddr.c:335:47: note: in expansion of macro ‘dns_opts’
  335 |         dill_dns_hosts, dill_dns_hints, NULL, dns_opts(), &rc);
      |                                               ^~~~~~~~
make[2]: *** [Makefile:1785: libdill_la-ipaddr.lo] Error 1
make[2]: Leaving directory '/hd-c/Downloads/tcp-client-server/libdill-2.14'
make[1]: *** [Makefile:1926: all-recursive] Error 1
make[1]: Leaving directory '/hd-c/Downloads/tcp-client-server/libdill-2.14'
make: *** [Makefile:21: libdill.a] Error 2

--- a/.gitignore
+++ b/.gitignore
@@ -1,0 +1,6 @@
+*.a
+*.mod
+*.o
+client
+server
+libdill

--- a/Makefile
+++ b/Makefile
@@ -17,11 +17,11 @@ client: client.f90 $(OBJS)
 server: server.f90 $(OBJS)
 	$(FC) $(FCFLAGS) $< -o $@ libdill.a -pthread
 
-libdill-2.14:
-	curl http://libdill.org/libdill-2.14.tar.gz | tar xz
+libdill:
+	git clone https://github.com/sustrik/libdill
 
-libdill.a: libdill-2.14
-	cd libdill-2.14 && ./configure && make && cp .libs/libdill.a ..
+libdill.a: libdill
+	cd libdill && ./autogen.sh && ./configure && $(MAKE) && cp .libs/libdill.a ..
 
 .f90.o:
 	$(FC) $(FCFLAGS) -c $<

--- a/README.md
+++ b/README.md
@@ -5,6 +5,11 @@ Companion code for Chapter 11 of [Modern Fortran: Building Efficient Parallel Ap
 
 It uses [libdill](http://libdill.org) as a sockets library.
 
+Please note that the install instructions from the book (downloading
+libdill-2.14) do not work with gfortran v9.x or later.
+The download and install procedure for libdill has been updated in this repo
+to allow building with the latest versions of gfortran.
+
 ## Getting started
 
 Download and build the code:

